src/miniscript/types/extra_props.rs

```diff
@@ -472,6 +639,31 @@ impl Property for ExtData {
                 (_, Some(x)) | (Some(x), _) => Some(x + 3),
                 (None, None) => None,
             },
+            stack_elem_count_sat: match (l.stack_elem_count_sat, r.stack_elem_count_sat) {
+                (Some(l), Some(r)) => Some(1 + cmp::max(l, r)),
+                (Some(l), None) => Some(1 + l),
+                (None, Some(r)) => Some(1 + r),
+                (None, None) => None,
+            },
+            stack_elem_count_dissat: match (l.stack_elem_count_dissat, r.stack_elem_count_dissat) {
+                (Some(l), Some(r)) => Some(1 + cmp::max(l, r)),
```

Comment:
This can be changed to unreachable!(), my bad!

Reply:
Yeah i understood your first comment (as in done in the current recursive function). However this still causes most tests to panics to do so.

Reply:
As discussed on IRC, this is incorrect for top-level scripts. We can keep this as it is for now without the panic

Reply:
I think this is fine. In non-malleable scripts this is actually unreachable; in malleable scripts (or incomplete script fragments) it is correct.

Reply:
Ah, I see, "correct" is maybe in the eye of the beholder. You could argue that, if both dissatisfactions are available, then we should actually report the minimum here rather than the maximum, otherwise we may incorrectly consider a script to be unsatisfiable.

But then we run into complicated issues ... there could be other reasons that we can't actually take one path, that we can't see from this part of the code. Maybe the dissatisfaction somehow requires a key/hash preimage that the user is missing; maybe it's blocked because of some other resource limit, or because it turns out to mix timelocks, or something crazy like that.

So best to return the maximum, and then when we're wrong, we'll say something is unsatisfiable when it's not...rather than the other way around.

And in any case, we'll be correct for non-malleable scripts, and users know that our analysis will not be very tight for malleable ones.